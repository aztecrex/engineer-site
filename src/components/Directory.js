import React from 'react';
import ArticleMenu from './ArticleMenu';

class Directory extends React.Component {

  render() {
    return (
      <div className="directory">
        <ArticleMenu directory={this.props.entries} />
      </div>
    );
  }

}

export {Directory};
