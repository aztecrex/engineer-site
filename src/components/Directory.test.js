import React from 'react';
import ezJson from 'enzyme-to-json';
import {Directory} from './Directory';

describe('unconnected', () => {
  it('renders', () => {
    // given
    const entries = [
      {id:'one',
      title:'One'}
    ];
    // when
    const rendered = shallow(<Directory entries={entries} />);

    //then
    expect(ezJson(rendered)).toMatchSnapshot();
  });
});
