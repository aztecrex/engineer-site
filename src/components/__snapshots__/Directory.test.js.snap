exports[`connected connects 1`] = `
<Directory
  dispatch={[Function]}
  entries={
    Array [
      Object {
        "published": 2016-12-04T00:00:00.000Z,
        "title": "four",
      },
      Object {
        "published": 2016-05-01T21:45:22.000Z,
        "title": "three",
      },
      Object {
        "published": 2016-05-01T21:45:21.100Z,
        "title": "two",
      },
      Object {
        "published": 2016-05-01T13:45:21.100Z,
        "title": "one",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  } />
`;

exports[`unconnected renders 1`] = `
<div
  className="directory">
  <ArticleMenu
    entries={
      Array [
        Object {
          "id": "one",
          "title": "One",
        },
      ]
    } />
</div>
`;
